src/rust/engine/process_execution/src/local.rs

```diff
@@ -361,27 +385,39 @@ impl CapturedWorkdir for CommandRunner {
       }
     }?;
 
-    debug!("spawned local process as {} for {:?}", child.id(), req);
+    debug!("spawned local process as {:?} for {:?}", child.id(), req);
     let stdout_stream = FramedRead::new(child.stdout.take().unwrap(), BytesCodec::new())
       .map_ok(|bytes| ChildOutput::Stdout(bytes.into()))
+      .fuse()
       .boxed();
     let stderr_stream = FramedRead::new(child.stderr.take().unwrap(), BytesCodec::new())
       .map_ok(|bytes| ChildOutput::Stderr(bytes.into()))
+      .fuse()
       .boxed();
-    let exit_stream = child
-      .into_stream()
-      .map_ok(|exit_status| {
-        ChildOutput::Exit(ExitCode(
-          exit_status
-            .code()
-            .or_else(|| exit_status.signal().map(Neg::neg))
-            .expect("Child process should exit via returned code or signal."),
-        ))
-      })
-      .boxed();
+    let owned_child = OwnedChildBuilder {
+      child: Box::new(child),
+      exit_stream_builder: |child: &mut Child| {
+        child
+          .wait()
```

Comment:
Here is the mutable borrow of `child`.

Reply:
Huh, using a closure like that works. I had not been wrapping it in an async block, which causes the borrow checker  to fail with:

```
   --> process_execution/src/local.rs:391:5
    |
376 |         child
    |         ----- `child` is borrowed here
...
391 | /     Ok(
392 | |       result_stream
393 | |         .map_err(|e| format!("Failed to consume process outputs: {:?}", e))
394 | |         .boxed(),
395 | |     )
    | |_____^ returns a value referencing data owned by the current function

error[E0515]: cannot return value referencing local variable `child`
```

The joys of Rust borrowing rules ... <sigh>

Updated to not use the `ouroboros` crate.